scripts/util_functions.sh

```diff
@@ -203,12 +206,66 @@ mount_partitions() {
     mount_name persist /persist
     if ! is_mounted /persist; then
       # Fallback to cache
-      mount_name cache /cache
+      mount_name "cache cac" /cache
       is_mounted /cache && PERSISTDIR=/cache || PERSISTDIR=
     fi
   fi
 }
 
+mount_apex() {
+  [ -d /system/apex ] || return
+  # APEX files present; need to extract and mount the payload imgs or if already extracted, mount folders
+  local APEX DEST LOOP MINORX NUM
+  [ -L /apex ] && rm -f /apex
+  [ -e /dev/block/loop1 ] && MINORX=$(ls -l /dev/block/loop1 | awk '{ print $6 }') || MINORX=1
+  NUM=0
+  for APEX in /system/apex/*; do
+    DEST=/apex/$(basename $APEX .apex)
+    [ "$DEST" == /apex/com.android.runtime.release ] && DEST=/apex/com.android.runtime
+    mkdir -p $DEST
+    case $APEX in
+      *.apex)
+        unzip -qo $APEX apex_payload.img -d /apex
+        mv -f /apex/apex_payload.img $DEST.img
+        mount -t ext4 -o ro,noatime $DEST.img $DEST 2>/dev/null
+        if [ $? != 0 ]; then
+          while [ $NUM -lt 64 ]; do
+            LOOP=/dev/block/loop$NUM;
+            (mknod $LOOP b 7 $((NUM * MINORX))
+            losetup $LOOP $DEST.img) 2>/dev/null
+            NUM=$((NUM + 1))
+            losetup $LOOP | grep -q $DEST.img && break
+          done
+          mount -t ext4 -o ro,loop,noatime $LOOP $DEST
+          if [ $? != 0 ]; then
+            losetup -d $LOOP 2>/dev/null
+          fi
+        fi
+      ;;
+      *) mount -o bind $APEX $DEST;;
+    esac
+  done
+  export ANDROID_RUNTIME_ROOT=/apex/com.android.runtime
+  export ANDROID_TZDATA_ROOT=/apex/com.android.tzdata
+  export BOOTCLASSPATH=/apex/com.android.runtime/javalib/core-oj.jar:/apex/com.android.runtime/javalib/core-libart.jar:/apex/com.android.runtime/javalib/okhttp.jar:/apex/com.android.runtime/javalib/bouncycastle.jar:/apex/com.android.runtime/javalib/apache-xml.jar:/system/framework/framework.jar:/system/framework/ext.jar:/system/framework/telephony-common.jar:/system/framework/voip-common.jar:/system/framework/ims-common.jar:/system/framework/android.test.base.jar:/apex/com.android.conscrypt/javalib/conscrypt.jar:/apex/com.android.media/javalib/updatable-media.jar
```

Comment:
com.android.runtime and com.android.conscrypt jars are enough, the rest can be removed

Reply:
You're wrong actually, tzdata and the TZDATA variable are absolutely required. I've tested this myself, directly, on a real device. Have you?

Reply:
I'm only talking on BOOTCLASSPATH, ANDROID_RUNTIME_ROOT and ANDROID_TZDATA_ROOT are needed. I didn't highlight them but github...

Reply:
It's just a path variable, and it's the one a booted ROM would have. No harm in leaving them in to leave the use-case of the script broad.